hbase-examples/src/main/java/org/apache/hadoop/hbase/coprocessor/example/row/stats/RowStatisticsCompactionObserver.java

```diff
@@ -0,0 +1,321 @@
+      RegionCoprocessorEnvironment regionEnv = (RegionCoprocessorEnvironment) e;
+      if (RowStatisticsUtil.isInternalTable(regionEnv)) {
+        return;
+      }
+      TableName tableName = regionEnv.getRegionInfo().getTable();
+      long tableCount = TABLE_COUNTERS.merge(tableName.getNameAsString(), -1L, Long::sum);
+      if (tableCount == 0) {
+        long regionCount = 0;
+        for (long count : TABLE_COUNTERS.values()) {
+          regionCount += count;
+        }
+        if (regionCount == 0) {
+          regionEnv.getMetricRegistryForRegionServer().remove(ROW_STATISTICS_DROPPED,
+            rowStatisticsDropped);
+          regionEnv.getMetricRegistryForRegionServer().remove(ROW_STATISTICS_PUT_FAILED,
+            rowStatisticsPutFailed);
+          boolean removed = regionEnv.getSharedData().remove(TABLE_RECORDER_KEY, tableRecorder);
+          if (removed) {
+            tableRecorder.close();
+          }
+        }
+      }
+    }
+  }
+
+  @Override
+  public void postStartMaster(ObserverContext<MasterCoprocessorEnvironment> ctx)
+    throws IOException {
+    try (Admin admin = ctx.getEnvironment().getConnection().getAdmin()) {
+      if (admin.tableExists(NAMESPACED_TABLE_NAME)) {
+        LOG.info("Table {} already exists. Skipping table creation process.",
+          NAMESPACED_TABLE_NAME);
+      } else {
+        boolean shouldCreateNamespace =
+          Arrays.stream(admin.listNamespaces()).filter(namespace -> namespace.equals(NAMESPACE))
+            .collect(Collectors.toUnmodifiableSet()).isEmpty();
+        if (shouldCreateNamespace) {
+          NamespaceDescriptor nd = NamespaceDescriptor.create(NAMESPACE).build();
+          try {
+            admin.createNamespace(nd);
+          } catch (IOException e) {
+            LOG.error("Failed to create namespace {}", NAMESPACE, e);
+          }
+        }
+        ColumnFamilyDescriptor cfd = ColumnFamilyDescriptorBuilder.newBuilder(CF).setMaxVersions(25)
+          .setTimeToLive((int) Duration.ofDays(7).toSeconds()).build();
+        TableDescriptor td =
+          TableDescriptorBuilder.newBuilder(NAMESPACED_TABLE_NAME).setColumnFamily(cfd).build();
+        LOG.info("Creating table {}", NAMESPACED_TABLE_NAME);
+        try {
+          admin.createTable(td);
+        } catch (IOException e) {
+          LOG.error("Failed to create table {}", NAMESPACED_TABLE_NAME, e);
+        }
+      }
+    } catch (IOException e) {
+      LOG.error("Failed to get Connection or Admin. Cannot determine if table {} exists.",
+        NAMESPACED_TABLE_NAME, e);
+    }
+  }
+
+  @Override
+  public InternalScanner preCompact(ObserverContext<RegionCoprocessorEnvironment> context,
+    Store store, InternalScanner scanner, ScanType scanType, CompactionLifeCycleTracker tracker,
+    CompactionRequest request) {
+    if (RowStatisticsUtil.isInternalTable(store.getTableName())) {
+      LOG.debug("Region {} belongs to an internal table {}, so no row statistics will be recorded",
```

Comment:
I think you can skip this log message. Internal tables are expected, so this will just add extra logging that doesn't really help an operator to understand what's happening in the system.